### PR TITLE
MAGECLOUD-3679: Add test to Cover BugFix from MAGECLOUD-3668

### PR DESCRIPTION
**Pull request by BaDos**
### Fixed Issues (if relevant)

https://magento2.atlassian.net/browse/MAGECLOUD-3679

### Manual testing scenarios

Need to perform a smoke testing `docker:build --mode=production` and `docker:build --mode=developer` and Docker as well

https://jira.corp.magento.com/browse/MAGECLOUD-27
https://jira.corp.magento.com/browse/MAGECLOUD-66
https://jira.corp.magento.com/browse/MAGECLOUD-112
https://jira.corp.magento.com/browse/MAGECLOUD-115
https://jira.corp.magento.com/browse/MAGECLOUD-50

### Contribution checklist
 - [ ] Pull request has a meaningful description of its purpose
 - [ ] All commits are accompanied by meaningful commit messages
 - [ ] All new or changed code is covered with unit/integration tests (if applicable)
 - [ ] All automated tests passed successfully (all builds on Travis CI are green)

### CODE REVIEW

File: src/Docker/Compose/ProductionCompose.php [hunk 1]
```diff
@@ -303,6 +324,22 @@ protected function getMagentoVolumes(bool $isReadOnly): array
         ];
     }
 
+    /**
+     * @param bool $isReadOnly
+     * @return array
+     */
+    protected function getMagentoBuildVolumes(bool $isReadOnly): array
```
Comment (shiftedreality) [inline src/Docker/Compose/ProductionCompose.php:331]: You should add `init` as well

File: tests/functional/Robo/Tasks/GenerateDockerCompose.php [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+
+    /**
+     * @var array
+     */
+    private $services;
+
+    /**
+     * @var array
+     */
+    private $availableServices = ['php', 'nginx', 'db', 'es', 'redis', 'rmq', 'node'];
```
Comment (shiftedreality) [inline tests/functional/Robo/Tasks/GenerateDockerCompose.php]: Use constants from `Service`

**Review by shiftedreality — CHANGES_REQUESTED** [review]

File: tests/functional/Robo/Tasks/GenerateDockerCompose.php (hunk 2 above)
Comment (BaDos) [reply tests/functional/Robo/Tasks/GenerateDockerCompose.php]: Done

File: src/Docker/Compose/ProductionCompose.php (hunk 1 above)
Comment (BaDos) [reply src/Docker/Compose/ProductionCompose.php:331]: It does not make sense.

Comment (shiftedreality) [reply src/Docker/Compose/ProductionCompose.php:331]: got it

**Review by shiftedreality — DISMISSED** [review]

**Review by billygilbert — DISMISSED** [review]

**Comment by andriyShevtsov** [comment]
QA approved

**Review by shiftedreality — APPROVED** [review]

**Review by billygilbert — APPROVED** [review]

**Comment by ghost** [comment]
Hi @BaDos, thank you for your contribution! </br>Please, complete [Contribution Survey](https://opensource.magento.com/survey/contribution/magento+ece-tools+539), it will take less than a minute. </br>Your feedback will help us to improve contribution process.

**Pull request merged**